include/deal.II/arborx/access_traits.h

```diff
@@ -0,0 +1,275 @@
+               dim == 2 ? 0 : static_cast<float>(min_corner[2])},
+              {static_cast<float>(max_corner[0]),
+               static_cast<float>(max_corner[1]),
+               dim == 2 ? 0. : static_cast<float>(max_corner[2])}};
+    }
+
+    using memory_space = Kokkos::HostSpace;
+  };
+
+
+
+  /**
+   * This struct allows ArborX to use std::vector<dealii::Point> as
+   * primitive.
+   */
+  template <int dim, typename Number>
+  struct AccessTraits<std::vector<dealii::Point<dim, Number>>, PrimitivesTag>
+  {
+    static_assert(dim != 1, "dim equal to one is not supported.");
+
+    /**
+     * Return the size of the vector @p v.
+     */
+    static std::size_t
+    size(const std::vector<dealii::Point<dim, Number>> &v)
+    {
+      return v.size();
+    }
+
+    /**
+     * Return an ArborX::Point from the dealii::Point v[i].
+     */
+    static Point
+    get(const std::vector<dealii::Point<dim, Number>> &v, std::size_t i)
+    {
+      // ArborX assumes that the point coordinates use float and that the point
+      // is 3D
+      return {{static_cast<float>(v[i][0]),
+               static_cast<float>(v[i][1]),
+               dim == 2 ? 0 : static_cast<float>(v[i][2])}};
+    }
+
+    using memory_space = Kokkos::HostSpace;
+  };
+
+
+
+  /**
+   * This struct allows ArborX to use PointIntersect in a predicate.
+   */
+  template <>
+  struct AccessTraits<dealii::ArborXWrappers::PointIntersect, PredicatesTag>
+  {
+    /**
+     * Number of Point stored in @p pt_intersect.
+     */
+    static std::size_t
+    size(const dealii::ArborXWrappers::PointIntersect &pt_intersect)
+    {
+      return pt_intersect.size();
+    }
+
+    /**
+     * Return an Arbox::intersects(ArborX::Point) object constructed from the
+     * ith dealii::Point stored in @p pt_intersect.
+     */
+    static auto
+    get(const dealii::ArborXWrappers::PointIntersect &pt_intersect,
+        std::size_t                                   i)
+    {
+      return intersects(Point{pt_intersect.points[i][0],
+                              pt_intersect.points[i][1],
+                              pt_intersect.points[i][2]});
+    }
+
+    using memory_space = Kokkos::HostSpace;
+  };
+
+
+
+  /**
+   * This struct allows ArborX to use BoundingBoxIntersect in a predicate.
+   */
+  template <>
+  struct AccessTraits<dealii::ArborXWrappers::BoundingBoxIntersect,
+                      PredicatesTag>
+  {
+    /**
+     * Number of BoundingBox stored in @p bb_intersect.
+     */
+    static std::size_t
+    size(const dealii::ArborXWrappers::BoundingBoxIntersect &bb_intersect)
+    {
+      return bb_intersect.size();
+    }
+
+    /**
+     * Return an Arbox::intersects(ArborX::Box) object constructed from the
+     * ith dealii::BoundingBox stored in @p bb_intersect.
+     */
+    static auto
+    get(const dealii::ArborXWrappers::BoundingBoxIntersect &bb_intersect,
+        std::size_t                                         i)
```

Comment:
It looks like this variable is unused. Maybe needs a `(void)i;` in the function definition?

Reply:
It is used at line 261

Reply:
Gosh, sorry about that. How unobservant of me :-/